server/middleware/populateCurrentUser.ts

```diff
@@ -8,9 +8,12 @@ export default function populateCurrentUser(userService: UserService): RequestHa
     try {
       if (req.user) {
         const callConfig = extractCallConfig(req)
-        const user = res.locals.user && (await userService.getUser(callConfig))
-        if (user) {
-          res.locals.user = { ...user, ...res.locals.user }
+        const userDetails = req.session.userDetails
+          ? req.session.userDetails
+          : res.locals.user && (await userService.getUser(callConfig))
+        if (userDetails) {
+          req.session.userDetails = userDetails
+          res.locals.user = { ...userDetails, ...res.locals.user }
```

Comment:
Up to you if you want to brave it, but I reckon this can be simplified a lot? I think a lot of this checking is redundant. Can it be?:

```
if (req.user) {
   const callConfig = extractCallConfig(req)
   const userDetails = req.session.userDetails || await userService.getUser(callConfig)
   req.session.userDetails = userDetails
   res.locals.user = { ...userDetails, ...res.locals.user }
}
```

Reply:
That is much prettier, I'll give it a try thanks :)

Reply:
Cool. Worth getting a test in for this? It was grandfathered in from the HMPPS Typescript Template without tests, but if we're making significant changes, might be worth testing

Reply:
Your version works great so I've switched to it. The old code was logging when there were no details so I've opted to keep it just in case it's ever useful for debugging.